lib/components/InputBase/withEnhancedInput.tsx

```diff
@@ -236,11 +238,13 @@ export function withEnhancedInput<IncomingProps = {}>(
 							</label>
 						)}
 					</NotchedBase>
-					{Boolean(hintText) && Boolean(hintText.length) && (
-						<HintText isActive={this.state.isActive}>
-							{hintText}
-						</HintText>
-					)}
+					{Boolean(hintText) &&
+						Boolean(hintText.length) &&
+						!disabled && (
```

Comment:
Think we can short-circuit here, ie check disabled first, because if that's false, v8 won't even bother executing the `Boolean` function.